File: tests/conformance/fixtures/typescript/20-edge-duplicate-after.ts
import { Pipeline } from '../../../../sdk/typescript/src/index';

const p = new Pipeline();

p.task('build').run('make build');

// Duplicate after calls should be deduplicated
p.task('test').run('go test').after('build').after('build');

p.task('deploy').run('./deploy.sh').after('build', 'test', 'build', 'test');

p.emit();

File: tests/conformance/fixtures/go/20-edge-duplicate-after.go
package main

import sykli "github.com/yairfalse/sykli/sdk/go"

func main() {
	p := sykli.New()

	p.Task("build").Run("make build")

	// Duplicate after calls should be deduplicated
	p.Task("test").Run("go test").After("build").After("build")

	p.Task("deploy").Run("./deploy.sh").After("build", "test", "build", "test")

	p.Emit()
}
